docs/contributing/contributor_guide.md

```diff
@@ -0,0 +1,121 @@
+# **Contributor's Guide for `metrics-utility`**
+
+## **1. Introduction**
+
+Welcome to the `metrics-utility` project! We appreciate your contributions, whether fixing bugs, improving documentation, or adding new features. This guide will help you get started with our contribution process.
+
+---
+
+## **2. Contribution Workflow**
+
+### **Forking Strategy**
+We use a **forking workflow** to ensure stability in the main repository. Follow these steps to contribute:
+
+1. **Fork the repository** to your GitHub account.
+2. **Clone your fork** to your local machine:
+   ```bash
+   git clone https://github.com/<your-username>/metrics-utility.git
+   cd metrics-utility
+   ```
+3. **Create a feature branch** in your fork:
+   ```bash
+   git checkout -b feature/<your-branch-name>
+   ```
+4. **Make changes** and commit them:
+   ```bash
+   git add .
+   git commit -m "Brief description of change"
+   ```
+   - If you are an internal contributor, ensure commits are **signed** (Verified tag).
+5. **Push your branch** to your fork:
+   ```bash
+   git push origin feature/<your-branch-name>
+   ```
+6. **Open a Pull Request (PR)** against the `devel` branch of the main repository.
+
+---
+
+## **3. What a PR Must Meet to Be Merged**
+To ensure consistency and maintainability, a PR should meet the following criteria:
+
+- ✅ Follow the **pull request template** (`.github/pull_request_template.md`).
+- ✅ Code should be properly **formatted and linted** using `pre-commit` and `ruff`.
+- ✅ Review the **essential linter settings** in [`pyproject.toml`](../pyproject.toml) to ensure compliance.
+- ✅ All **conversations on the PR must be resolved** before merging.
+- ✅ PR must receive **at least two approvals** from maintainers.
+- ✅ Internal contributors must **sign commits** (Verified tag).
+- ✅ PR must **pass all required checks**, including static analysis and pre-commit hooks.
+
+For **external contributors**, a member of the `aap-metrics-write` group will review and merge your PR since direct access is restricted.
+
+---
+
+## **4. Setting Up Your Development Environment**
```

Comment:
May want to mention somewhere here that dependency changes require uv sync. It's in the developer setup but could also be mentioned here